sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -78,11 +82,14 @@ trait NonLeafStatementExec extends CompoundStatementExec {
         case Array(field) if field.dataType == BooleanType =>
           df.limit(2).collect() match {
             case Array(row) => row.getBoolean(0)
-            case _ => false
+            case _ =>
+              throw QueryExecutionErrors.invalidBooleanStatementError(statement.getText)
           }
-        case _ => false
+        case _ =>
+          throw QueryExecutionErrors.invalidBooleanStatementError(statement.getText)
       }
-    case _ => false
+    case _ =>
+      throw SparkException.internalError("Boolean condition must be SingleStatementExec")
```

Comment:
@cloud-fan do we want to return internal errors in such cases? this is not expected to be hit ever, so I guess it's fine since it would mean we have a bug.

Reply:
Does this mean the boolean query returns more than one column? It seems not possible to happen and internal error LGTM.